### PR TITLE
introduce output_transform to Metrics

### PR DESCRIPTION
**Pull request by jasonkriss**

### CODE REVIEW

File: ignite/metrics/metric.py [hunk 1]
```diff
@@ -11,8 +11,15 @@ class Metric(object):
...
+
+    Args:
+        output_transform (callable): an optional callable that used to transform
+            the model's output into the form expected by the metric. This can be
+            useful if, for example, you have a multi-output model and you want to
+            compute the metric with respect to one of the outputs.
     """
-    def __init__(self):
+    def __init__(self, output_transform=None):
+        self._output_transform = output_transform
```
Comment (vfdev-5) [inline ignite/metrics/metric.py]: nit: maybe simlier to define self._output_transform as `lambda x: x` if no output_transform provided and avoid 
```
output = self._output_transform(state.output) if self._output_transform else state.output
```
in the loop

Comment (jasonkriss) [reply ignite/metrics/metric.py]: yea good point. I was considering that. Let's do that. It's what we did before with the old `History`.

**Comment by alykhantejani** [comment]
lets merge #123 first and then this will need a small update

**Comment by jasonkriss** [comment]
@alykhantejani Should be all good once Travis is done.

**Comment by jasonkriss** [comment]
@alykhantejani 👍 #131

**Pull request merged**